File: FE/DATN-Fashion-Shop-FE/src/app/models/Product/Attribute.ts
import { BaseEntity} from '../BaseEntity';
import { AttributeValue} from './AttributeValue';

export class Attribute extends BaseEntity {
  id: number;
  name: string;
  attributeValues: AttributeValue[];

  constructor(data?: Partial<Attribute>) {
    super(data);
    this.id = data?.id || 0;
    this.name = data?.name || '';
    this.attributeValues = data?.attributeValues?.map(av => new AttributeValue(av)) || [];
  }
}

File: DATN-Fashion-Shop-BE/src/main/java/com/example/DATN_Fashion_Shop_BE/model/Attribute.java
package com.example.DATN_Fashion_Shop_BE.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "attributes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Attribute extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", length = 255, nullable = false)
    private String name;

    @OneToMany(mappedBy = "attribute", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<AttributeValue> attributeValues = new HashSet<>();
}
